### PR TITLE
Implemented autoreload on code changes in dockerfile

### PR DESCRIPTION
**Pull request by leothomas**
### What I'm changing: 

Stopping and re-starting the API docker container every time I wanted to test some changes was becoming time consuming, so I implement autoreloading on file changes. 

### How I did it: 
I added `CMD ["/start-reload.sh"]` to the `/ecs/Dockerfile` and mounted the `covid_api/` code directory in the docker container (in `docker-compose.yml`)

### How you can test it: 
Run `docker-compose up --build` and wait for the API to get up and running. 

Make a small code modification in the API files (eg: add a print statement to the `/datasets` LIST endpoint) and verify that the API reloads: 
image

### Notes: 
Was there a reason this was not enabled initially? I want to be sure this won't cause any issues with the production deployments of the API. @drewbo @olafveerman ?

### CODE REVIEW

**Comment by drewbo** [comment]
I was generally running without docker and that process autoreloads already. This seems fine but verify with @vincentsarago who knows the most about FastAPI docker

**Comment by drewbo** [comment]
Also it shouldn't affect production at all because the change is in the ECS Dockerfile (and production uses lambda)

**Review by drewbo — APPROVED** [review]

**Comment by vincentsarago** [comment]
@leothomas did you test it ? what I understand is you:
1. copy app/ and setup.py to the docker images 
2. install with `-e` 

I don't think any change in your local directory will be applied to the app itself.

File: Dockerfiles/ecs/Dockerfile [hunk 1]
```diff
@@ -10,3 +10,4 @@ COPY covid_api/ /app/covid_api/
 COPY setup.py /app/setup.py
 
 RUN pip install -e /app/. --no-cache-dir
+CMD ["/start-reload.sh"]
```
Comment (vincentsarago) [inline Dockerfiles/ecs/Dockerfile:13]: ref https://github.com/tiangolo/uvicorn-gunicorn-fastapi-docker#usage

**Comment by vincentsarago** [comment]
@leothomas I don't think it will break anything because it's not used in production 🤷‍♂️ 

FYI: locally I'm usually doing `uvicorn titiler.main:app --reload` but, yes you have to have everything installed

**Review by vincentsarago — APPROVED** [review]

**Comment by leothomas** [comment]
@vincentsarago changes in the code directory definitely trigger a reload of the API. Is that unexpected, given the `-e` flag in the pip install?

**Comment by vincentsarago** [comment]
@leothomas 🤷‍♂️ seems weird to be, the `-e` should tell python to link the code hosted in `docker:app/` directory, not sure what is the magic of the `/start-reload.sh`, but if it works for you go ahead!

**Comment by leothomas** [comment]
Awesome thanks everyone!

**Pull request merged**